gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/ReferralRequestMapper.java

```diff
@@ -274,4 +287,44 @@ private boolean isNotSelfReferral(RCMRMT030101UKRequestStatement requestStatemen
         }
         return true;
     }
+
+    private static void populateSupportingInfoWithDocumentReferencesFromLinkSets(
+        RCMRMT030101UKEhrExtract ehrExtract,
+        List<ReferralRequest> referralRequests
+    ) {
+        extractReferralRequestIdToDocumentReferences(ehrExtract).forEach(
+            (referralRequestId, documentReferences) -> referralRequests.stream()
+                .filter(referralRequest -> referralRequestId.equals(referralRequest.getId()))
+                .findFirst()
+                .ifPresent(referencedReferralRequest -> referencedReferralRequest.setSupportingInfo(documentReferences))
+        );
+    }
+
+    private static Map<String, List<Reference>> extractReferralRequestIdToDocumentReferences(
```

Comment:
Just thinking about this method, and edge cases.

There's nothing stopping us being given some XML which results in:
```
"123" => [a, b],
"456" => [c, d],
"123" => [c],
```

I don't think I have a clear answer as to what the adaptor should do in this situation, but I'm throwing this out there in case you have thoughts. I'm slightly worried that the `collect` method here might choke on this, whereas really it should at least do _something_.

Reply:
Have added the functionality for this and will handle the merge correctly for the cases above, will also address a case when:

```text
"123" => [a, b]
"123" => [a, c]
```
to ensure that the same reference is not duplicated.

PseudoXML:
```xml
<LinkSet>
  <id root="x">
  <namedStatement>
    <ref id="123" />
  </namedStatement>
  <component ref="a" />
  <component ref="b" />
</LinkSet>

<LinkSet>
  <id root="y">
  <namedStatement>
    <ref id="123" />
  </namedStatement>
 <component ref="a" />
  <component ref="c" />
</LinkSet>
```